test/bench/synthetic_benchmarks.cpp

```diff
@@ -0,0 +1,241 @@
+bytecode generate_loop_inner_code(CodeParams params)
+{
+    const auto [opcode, mode] = params;
+    const auto kind = get_instruction_kind(opcode);
+    switch (mode)
+    {
+    case Mode::min_stack:
+        switch (kind)
+        {
+        case InstructionKind::push:
+            // PUSH1 POP PUSH1 POP ...
+            return stack_limit * (push(opcode, {}) + OP_POP);
+        case InstructionKind::swap:
+        {
+            const auto n = opcode - OP_SWAP1 + 1;
+            // DUP1 SWAP1 SWAP1 ... POP
+            return n * OP_DUP1 + stack_limit * 2 * bytecode{opcode} + n * OP_POP;
+        }
+        case InstructionKind::dup:
+        {
+            const auto n = opcode - OP_DUP1;
+            // DUP1 DUP1 POP DUP1 POP ... POP
+            return n * OP_DUP1 + (stack_limit - n) * (bytecode{opcode} + OP_POP) + n * OP_POP;
+        }
+        case InstructionKind::nullop:
+            // CALLER POP CALLER POP ...
+            return stack_limit * (bytecode{opcode} + OP_POP);
+        case InstructionKind::nop:
+            // JUMPDEST JUMPDEST ...
+            return stack_limit * 2 * bytecode{opcode};
+        case InstructionKind::unop:
+            // DUP1 NOT NOT ... POP
+            return OP_DUP1 + stack_limit * 2 * bytecode{opcode} + OP_POP;
+        case InstructionKind::binop:
+            // DUP1 DUP1 ADD DUP1 ADD DUP1 ADD ... POP
+            return OP_DUP1 + (stack_limit - 1) * (OP_DUP1 + bytecode{opcode}) + OP_POP;
+        default:
+            break;
+        }
+        break;
+    case Mode::full_stack:
+        switch (kind)
+        {
+        case InstructionKind::push:
+            // PUSH1 PUSH1 PUSH1 ... POP POP POP ...
+            return stack_limit * push(opcode, {}) + stack_limit * OP_POP;
+        case InstructionKind::dup:
+        {
+            const auto n = opcode - OP_DUP1;
+            // DUP1 DUP1 POP DUP1 POP ... POP
+            return n * OP_DUP1 + (stack_limit - n) * bytecode{opcode} + stack_limit * OP_POP;
+        }
+        case InstructionKind::nullop:
+            // CALLER CALLER ... POP POP ...
+            return stack_limit * opcode + stack_limit * OP_POP;
+        case InstructionKind::binop:
+            // DUP1 DUP1 DUP1 ... ADD ADD ADD ... POP
+            return stack_limit * OP_DUP1 + (stack_limit - 1) * opcode + OP_POP;
+        default:
+            break;
+        }
+        break;
+    }
+
+    return {};
+}
+
+/// Generates a benchmark loop with given inner code.
+///
+/// This generates do-while loop with 255 iterations and it starts with PUSH1 of 255 as the loop
+/// counter. The while check is done as `(counter += -1) != 0`. The SUB is avoided because it
+/// consumes arguments in unnatural order and additional SWAP would be required.
+///
+/// The loop counter stays on the stack top. The inner code is allowed to duplicate it, but must not
+/// modify it.
+bytecode generate_loop_v1(const bytecode& inner_code)
+{
+    return push(255) + OP_JUMPDEST + inner_code +
```

Comment:
Perhaps you want to make the first push explicit: `push(uint8_t{255})` so that the jump offset `push(2)` remains always valid.